framework/serialization/providers/filedataprovider.cpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
```

Comment:
#pragma once doesn't seem to make sense to me in cpp files (and gcc complains) - maybe remove these?